File: backend/src/main/java/com/whiteblue/backend/domain/application/ApplicationService.java
package com.whiteblue.backend.domain.application;

import com.whiteblue.backend.domain.application.DTO.GetApplicationResponse;
import com.whiteblue.backend.domain.application.DTO.SaveApplicationRequest;
import com.whiteblue.backend.domain.user.User;
import com.whiteblue.backend.domain.user.UserRepository;
import com.whiteblue.backend.security.oAuth.OAuthUser;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Transactional
@Service
public class ApplicationService {
    private final ApplicationRepository applicationRepository;

    private final UserRepository userRepository;

    public GetApplicationResponse findByUser(OAuthUser oAuthUser) {
        User user = userRepository.findById(oAuthUser.getId())
                                  .orElseThrow();
        Application application = applicationRepository.findByWriterId(user.getId())
                                                       .orElseGet(() -> applicationRepository.save(Application.builder()
                                                                                                              .name("")
                                                                                                              .phoneNumber("")
                                                                                                              .major("")
                                                                                                              .introduction("")
                                                                                                              .writer(user)
                                                                                                              .build()));

        return GetApplicationResponse.builder()
                                     .name(application.getName())
                                     .phoneNumber(application.getPhoneNumber())
                                     .major(application.getMajor())
                                     .introduction(application.getIntroduction())
                                     .writer(user)
                                     .build();
    }

    public void save(SaveApplicationRequest saveApplicationRequest, OAuthUser oAuthUser) {
        User user = userRepository.findById(oAuthUser.getId())
                                  .orElseThrow();

        Application application = applicationRepository.findByWriterId(oAuthUser.getId())
                                                       .orElse(applicationRepository.save(saveApplicationRequest.toEntity()));
        application.setName(saveApplicationRequest.getName());
        application.setPhoneNumber(saveApplicationRequest.getPhoneNumber());
        application.setMajor(saveApplicationRequest.getMajor());
        application.setIntroduction(saveApplicationRequest.getIntroduction());
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/application/ApplicationService.kt
package com.whiteblue.backend.domain.application

import com.whiteblue.backend.domain.application.dto.GetApplicationResponse
import com.whiteblue.backend.domain.application.dto.SaveApplicationRequest
import com.whiteblue.backend.domain.user.UserRepository
import com.whiteblue.backend.security.oAuth.OAuthUser
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class ApplicationService(
    private val applicationRepository: ApplicationRepository,

    private val userRepository: UserRepository
) {
    @Transactional(readOnly = true)
    fun findByUser(oAuthUser: OAuthUser): GetApplicationResponse? {
        return applicationRepository.findByWriterId(oAuthUser.id)
            ?.run {
                GetApplicationResponse(
                    name = name,
                    phoneNumber = phoneNumber,
                    major = major,
                    introduction = introduction,
                    writer = writer
                )
            }
    }

    fun save(saveApplicationRequest: SaveApplicationRequest, oAuthUser: OAuthUser) {
        applicationRepository.findByWriterId(oAuthUser.id)
            ?.apply {
                name = saveApplicationRequest.name
                phoneNumber = saveApplicationRequest.phoneNumber
                major = saveApplicationRequest.major
                introduction = saveApplicationRequest.introduction
            } ?: applicationRepository.save(
            Application(
                name = "",
                phoneNumber = "",
                major = "",
                introduction = "",
                writer = userRepository.findByOAuthUser(oAuthUser)
            )
        )
    }
}
